go-controller/pkg/util/util.go

```diff
@@ -313,3 +313,15 @@ func UpdateNodeSwitchExcludeIPs(nbClient libovsdbclient.Client, nodeName string,
 
 	return nil
 }
+
+// ValidateAndGetIPString validates the IP address and returns the IP in
+// string format removing any leading zeros.
+func ValidateAndGetIPString(ip string) (string, error) {
```

Comment:
If you are only using this function for IPs in kubernetes API objects, then those fields have already been validated, so you know they contain IPs that are valid according to `net.ParseIP`, and it's just a matter of removing the leading zeros. In which case, you don't need to return an `error`.